tests/integration/server/proposal_api_tests.py

```diff
@@ -0,0 +1,378 @@
+        usernames_for_deleting.append(manager_payload["username"])
+        manager_id = create_user(session, manager_payload)
+        expected_approver_ids.append(manager_id)
+        i = 0
+        while i < 4:
+            manager_payload = create_test_manager_payload()
+            manager_payload["manager"] = manager_id
+            manager_id = create_user(session, manager_payload)
+            expected_approver_ids.append(manager_id)
+            usernames_for_deleting.append(manager_payload["username"])
+            i += 1
+        roleowner_payload["manager"] = manager_id
+        roleowner_id = create_user(session, roleowner_payload)
+        expected_approver_ids.append(roleowner_id)
+        new_roleowner_payload = {
+            "name": "newroleowner_name",
+            "username": "newroleowner_username",
+            "password": "newroleowner_password",
+            "email": "newroleowner@email_id",
+        }
+        new_manager_payload = create_test_manager_payload()
+        usernames_for_deleting.append(new_manager_payload["username"])
+        manager_id = create_user(session, new_manager_payload)
+        expected_approver_ids.append(manager_id)
+        i = 0
+        while i < 4:
+            new_manager_payload = create_test_manager_payload()
+            new_manager_payload["manager"] = manager_id
+            manager_id = create_user(session, new_manager_payload)
+            expected_approver_ids.append(manager_id)
+            usernames_for_deleting.append(new_manager_payload["username"])
+            i += 1
+        new_roleowner_payload["manager"] = manager_id
+        new_roleowner_id = create_user(session, new_roleowner_payload)
+        expected_approver_ids.append(new_roleowner_id)
+        role_payload = {"name": "Test_role"}
+        role_payload["owners"] = roleowner_id
+        role_payload["administrators"] = roleowner_id
+        role_id = create_role(session, role_payload)
+        add_role_owner_payload = {
+            "id": new_roleowner_id,
+            "reason": "for testing",
+            "metadata": "",
+        }
+        response = add_role_owner(session, add_role_owner_payload, role_id)
+        update_proposal_payload = {"status": "APPROVED", "reason": "test"}
+        proposal_update_response = update_propopsal(
+            session, update_proposal_payload, response.json()["proposal_id"]
+        )
+        user_payload = {
+            "name": "user_name",
+            "username": "user_username",
+            "password": "user_password",
+            "email": "user@email_id",
+        }
+        user_id = create_user(session, user_payload)
+        creating_proposal_payload = {"id": user_id, "reason": "add the test member"}
+        proposal_id = create_proposal(session, creating_proposal_payload, role_id)
+        proposal_information = get_proposal_details(session, proposal_id)
+        i = 0
+        while i < len(expected_approver_ids):
+            assert (
```

Comment:
As we discussed please try to write the tests in a way that doesn't require an assertion in a while loop and preferably has one assertion per function.